### PR TITLE
feat: refactor drawer component

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?
Refactors sheet to be more structured to allow us to use forms easily inside of it. Instead of hacking the submission we make it actually be able to be wrapped in a form. 

Availability edit 
CleanShot 2024-07-24 at 13 50 00

Edit User Sheet: 
CleanShot 2024-07-24 at 13 50 13
CleanShot 2024-07-24 at 13 50 16

## Mandatory Tasks (DO NOT REMOVE)

- [X] I have self-reviewed the code (A decent size PR without self-review might be rejected).

## How should this be tested?
Open members edit 
or availability edit on timezone buddy

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/ui/components/sheet/sheet.tsx [hunk 1]
```diff
@@ -1,215 +1,173 @@
...
-      size: {
-        content: "",
-        default: "",
-        sm: "",
-        lg: "",
-        xl: "",
-        full: "",
-      },
-    },
-    compoundVariants: [
```
Comment (sean-brydon) [inline packages/ui/components/sheet/sheet.tsx:70]: These were never used. Lets simplify this a lot

**Review by PeerRich — DISMISSED** [review]

**Comment by keithwillcode** [comment]
@anikdhabal @Amit91848 One of you able to test this? Would like to ship today if possible

**Comment by anikdhabal** [comment]
> @anikdhabal @Amit91848 One of you able to test this? Would like to ship today if possible

Let me test

**Comment by anikdhabal** [comment]
Screenshot 2024-07-25 141028
Need to be fixed

**Comment by sean-brydon** [comment]
> Screenshot 2024-07-25 141028 Need to be fixed

Hey ermm can you be a bit more specific on what needs fixed here? Do you mean the padding on the bottom? Can't tell from the SC

**Comment by sean-brydon** [comment]
> Screenshot 2024-07-25 141028 Need to be fixed



> > Hey ermm can you be a bit more specific on what needs fixed here? Do you mean the padding on the bottom? Can't tell from the SC
> 
> Yeah, the bottom padding.
CleanShot 2024-07-25 at 10 17 30
Fixed

**Comment by anikdhabal** [comment]
Screenshot 2024-07-25 150029
@sean-brydon need equal padding on x. This is broken after you commit

**Review by anikdhabal — APPROVED** [review]
Tested it, LGTM!

**Pull request merged**